### DISCUSSION TITLE
`github-release-attachments` datasource always fails with `Datasource unauthorized`

### DISCUSSION THREAD

**Discussion by gandro** [renovatebot/renovate#21231] [Request Help] [answered]
### I'm using the Mend Renovate hosted app on github.com.

- [X] I'm using the Mend Renovate hosted app.

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe your problem.

The `github-release-attachments` datasource introduced in #20178 as a drop-in replacement for `github-releases` when updating attachment digests of GitHub releases via Regex manger is not functional. Whenever Renovate v35 tries to fetch release attachments via GitHub API, it fails with an Unauthorized error, _even though the GitHub token is valid_.

This happens on the Renovate GitHub App, but to further troubleshoot this, I've also tested this with self-hosted Renovate v34 vs v35.
 
```
DEBUG: Datasource unauthorized (repository=gandro/renovate-github-release-attachments-bug-repro)
       "datasource": "github-release-attachments",
       "packageName": "cilium/hubble",
       "url": "https://api.github.com/graphql"
```
 
Where as in Renovate v34 using the `github-releases` datasource with the old behavior, the same Renovate configuration (and GitHub token) works just fine:
 
```
DEBUG: getDigest (repository=gandro/renovate-github-release-attachments-bug-repro)
       "repo": "cilium/hubble",
       "currentValue": "v0.11.2",
       "currentDigest": "fa2b5a1468f17957899063ad2ff1b51c68aae955a2b5d09390174a7b36a80bdb",
       "registryUrl": "https://github.com",
       "newValue": "v0.11.3"
```

I suspect because the newly added `github-release-attachments` has not been hooked up with the GitHub API token internally. For example, I noticed it is not listed in `GITHUB_API_USING_HOST_TYPES` in `lib/constants/plattform.ts`. I've also noticed that sometimes, the request seems to work if there was a prior request initiated by the `github-releases` data source, but it never works if the `github-release-attachments` data source is used in isolation.

Reproduction repository can be found here: https://github.com/gandro/renovate-github-release-attachments-bug-repro

Previously reported as an issue in https://github.com/renovatebot/renovate/issues/21230

### Relevant debug logs.

<details><summary>Renovate v35 (with `github-release-attachments- as data source)</summary>

```
DEBUG: Found regex package files (repository=gandro/renovate-github-release-attachments-bug-repro)
DEBUG: Found 1 package file(s) (repository=gandro/renovate-github-release-attachments-bug-repro)
 INFO: Dependency extraction complete (repository=gandro/renovate-github-release-attachments-bug-repro, baseBranch=main)
       "stats": {
         "managers": {"regex": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
DEBUG: POST https://api.github.com/graphql = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=229) (repository=gandro/renovate-github-release-attachments-bug-repro)
DEBUG: Datasource unauthorized (repository=gandro/renovate-github-release-attachments-bug-repro)
       "datasource": "github-release-attachments",
       "packageName": "cilium/hubble",
       "url": "https://api.github.com/graphql"
DEBUG: Failed to look up github-release-attachments package cilium/hubble (repository=gandro/renovate-github-release-attachments-bug-repro, packageFile=README.md, dependency=cilium/hubble)
```

Full logs here: https://gist.github.com/gandro/1e0167205049c9464db578b89c3718f3

</details>

<details><summary>Renovate v34 (with `github-releases` as data source)</summary>

```
DEBUG: Found regex package files (repository=gandro/renovate-github-release-attachments-bug-repro)
DEBUG: Found 1 package file(s) (repository=gandro/renovate-github-release-attachments-bug-repro)
 INFO: Dependency extraction complete (repository=gandro/renovate-github-release-attachments-bug-repro, baseBranch=main)
       "stats": {
         "managers": {"regex": {"fileCount": 1, "depCount": 1}},
         "total": {"fileCount": 1, "depCount": 1}
       }
DEBUG: getDigest (repository=gandro/renovate-github-release-attachments-bug-repro)
       "repo": "cilium/hubble",
       "currentValue": "v0.11.2",
       "currentDigest": "fa2b5a1468f17957899063ad2ff1b51c68aae955a2b5d09390174a7b36a80bdb",
       "registryUrl": "https://github.com",
       "newValue": "v0.11.3"
DEBUG: getDigest (repository=gandro/renovate-github-release-attachments-bug-repro)
       "repo": "cilium/hubble",
       "currentValue": "v0.11.2",
       "currentDigest": "fa2b5a1468f17957899063ad2ff1b51c68aae955a2b5d09390174a7b36a80bdb",
       "registryUrl": "https://github.com",
       "newValue": "v0.11.2"
DEBUG: PackageFiles.add() - Package file saved for base branch (repository=gandro/renovate-github-release-attachments-bug-repro, baseBranch=main)
DEBUG: Package releases lookups complete (repository=gandro/renovate-github-release-attachments-bug-repro, baseBranch=main)
```

Full logs here: https://gist.github.com/gandro/adc2e57ef0378908f2b8fe25f15622bf

</details>

**Comment by rarkins** [comment, marked as answer]
Reproduction confirmed, initial bug report reopened
